py-polars/src/conversion.rs

```diff
@@ -656,10 +661,14 @@ impl<'s> FromPyObject<'s> for Wrap<AnyValue<'s>> {
                         let (seconds, microseconds) = {
                             let datetime = PyModule::import(py, "datetime")?;
                             let timezone = datetime.getattr("timezone")?;
-                            let kwargs = PyDict::new(py);
-                            kwargs.set_item("tzinfo", timezone.getattr("utc")?)?;
-                            let dt = ob.call_method("replace", (), Some(kwargs))?;
-
+                            let tzinfo = ob.getattr("tzinfo")?;
+                            let dt = if tzinfo.is_none() {
+                                let kwargs = PyDict::new(py);
+                                kwargs.set_item("tzinfo", timezone.getattr("utc")?)?;
+                                ob.call_method("replace", (), Some(kwargs))?
+                            } else {
+                                ob
+                            };
```

Comment:
the idea's that instead of
1. forcing to UTC
2. replacing the original time zone

to
1. keep the original time zone (and force to UTC if there isn't one)
2. convert back to the original time zone